specification/datalake-store/resource-manager/Microsoft.DataLakeStore/stable/2016-11-01/account.json

```diff
@@ -630,12 +630,235 @@
         },
         "x-ms-examples": {
-          "Deletes the specified firewall rule from the specified Data Lake Store account": {
+          "Deletes the specified firewall rule from the specified Data Lake Store account.": {
             "$ref": "./examples/FirewallRules_Delete.json"
           }
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataLakeStore/accounts/{accountName}/virtualNetworkRules": {
+      "get": {
+        "tags": [
+          "VirtualNetworkRules"
+        ],
+        "operationId": "VirtualNetworkRules_ListByAccount",
+        "description": "Lists the Data Lake Store virtual network rules within the specified Data Lake Store account.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AccountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the list of virtual network rules.",
+            "schema": {
+              "$ref": "#/definitions/VirtualNetworkRuleListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Lists the Data Lake Store virtual network rules within the specified Data Lake Store account.": {
+            "$ref": "./examples/VirtualNetworkRules_ListByAccount.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataLakeStore/accounts/{accountName}/virtualNetworkRules/{virtualNetworkRuleName}": {
+      "put": {
+        "tags": [
+          "VirtualNetworkRules"
+        ],
+        "operationId": "VirtualNetworkRules_CreateOrUpdate",
+        "description": "Creates or updates the specified virtual network rule. During update, the virtual network rule with the specified name will be replaced with this new virtual network rule.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AccountNameParameter"
+          },
+          {
+            "name": "virtualNetworkRuleName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the virtual network rule to create or update."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CreateOrUpdateVirtualNetworkRuleParameters"
```

Comment:
The same model can be used between request and response by marking the appropriate properties as readonly if they are not allowed to be set in the request but are present in response.

Reply:
@ravbhatnagar This is surely one way to design this.  The design we took, however, was this type of model for all API requests.  Changing it to your suggested model would create inconsistency.  This is something that @lewu-msft can consider for a future design revamp.

Reply:
It makes usability difficult from a client point of view. It makes the GET-PUT pipeline harder from the clients. I am fine with this if the SDK team is ok with this. @anuchandy FYI